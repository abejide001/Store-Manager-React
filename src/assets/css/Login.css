@import url("https://fonts.googleapis.com/css?family=Charmonman|Roboto|K2D");
#login-area {
  min-height: 510px;
  font-family: "K2D", sans-serif;
  flex: 1;
}
.login-showcase {
  display:  flex;
  flex-direction: column;
  height: 100%;
}
.login-body {
  position: absolute;
  width: 100%;
  top:0;
  bottom: 0;
}
.login-box {
  width: 280px;
  position: absolute;
  top: 25%;
  left: 50%;
  transform: translate(-50%, -50%);
  margin-top: 100px;
}

.login-box h1 {
  font-size: 40px;
  border-bottom: 2px solid gray;
}

.login-textbox {
  width: 100%;
  font-size: 30px;
  overflow: hidden;
  padding: 8px 0px;
  margin: 8px 0px;
  border-bottom: 1px solid lightcoral;
}

.login-textbox input {
  border: none;
  outline: none;
  background: none;
  color: black;
  font-size: 20px;
  float: left;
  width: 80%;
  font-family: "K2D", sans-serif;
}

.login-textbox i {
  width: 40px;
  float: left;
  text-align: center;
  font-size: 24px;
}
.alert {
  display: none;
}
.alert.error .inner {
  border: 2px solid rgb(238,211,215);
  background-color: rgb(242,222,222);
  font-size: 20px;
}

.alert.error .inner,.alert.error .close {
  color: rgb(185,74,72);
}

@media (max-width: 348px) {
  .login-box h1 {
    font-size: 25px;
  }
  .login-box {
    margin-top: 150px;
  }
}
@media (max-width: 769px) {
  .login-box h1 {
    font-size: 25px;
    text-align: center;
  }
  .login-box {
    margin-top: 150px;
  }
  #login-area  {
    min-height: 850px;
  }
}