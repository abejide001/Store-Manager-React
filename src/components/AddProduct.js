/* eslint-disable camelcase */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { addProducts } from '../actions/productActions';
import logo from '../assets/images/logo.png';
import '../assets/css/AddProduct.css';

export class AddProduct extends Component {
  state = {
    name: '',
    price: '',
    quantity_in_inventory: '',
    product_image: '',
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const {
      name, price, quantity_in_inventory, product_image,
    } = this.state;
    const productData = {
      name,
      price,
      quantity_in_inventory,
      product_image,
    };
    this.props.addProducts(productData);
  };

  handleChange = (e) => {
    this.setState({ [e.currentTarget.name]: e.currentTarget.value });
  };

  render() {
    const {
      name, price, quantity_in_inventory, product_image,
    } = this.state;
    return (
      <div>
        <header>
          <div className="container">
            <div id="header-title">
              <Link to="/">
                <img src={logo} alt="logo" />
              </Link>
            </div>
            <nav>
              <ul>
                <li>
                  <Link to="/admin">Admin</Link>
                </li>
              </ul>
            </nav>
          </div>
        </header>
        <section id="create-product">
          <form className="submission-form" onSubmit={this.handleSubmit}>
            <h1>Create Product</h1>
            <input
              type="text"
              name="name"
              placeholder="Product Name"
              value={name}
              onChange={this.handleChange}
              required
            />
            <input
              type="number"
              name="price"
              placeholder="Price"
              onChange={this.handleChange}
              value={price}
              required
            />
            <input
              type="number"
              name="quantity_in_inventory"
              value={quantity_in_inventory}
              onChange={this.handleChange}
              placeholder="quantity in inventory"
              required
            />
            <input
              type="url"
              name="product_image"
              value={product_image}
              placeholder="Image URL"
              onChange={this.handleChange}
              required
            />
            <input type="submit" value="CREATE PRODUCT" id="sendBtn" />
          </form>
        </section>
      </div>
    );
  }
}
AddProduct.propTypes = {
  addProducts: PropTypes.func.isRequired,
};
const mapStateToProps = state => ({
  product: state.product,
});
export default connect(mapStateToProps, { addProducts })(AddProduct);
