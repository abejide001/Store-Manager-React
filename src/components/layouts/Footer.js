import React from 'react';
import '../../assets/css/style.css';

export default function Footer() {
  return (
    <div>
      <footer>
        <div className="container">
          <h3>
            Store Manager
            {' '}
            {new Date().getFullYear()}
          </h3>
        </div>
      </footer>
    </div>
  );
}
