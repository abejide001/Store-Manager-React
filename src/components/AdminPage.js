/* eslint-disable jsx-a11y/alt-text */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getProducts, deleteProduct } from '../actions/productActions';
import logo from '../assets/images/logo.png';
import '../assets/css/Admin.css';
import '../assets/css/style.css';
import Footer from './layouts/Footer';
import Spinner from '../common/Spinner';

export class Admin extends Component {
  componentWillMount() {
    this.props.getProducts();
  }

handleDelete = (id) => {
  this.props.deleteProduct(id);
}

render() {
  const { products } = this.props.product.products;
  return (
    <div>
      <header>
        <div className="container">
          <div id="header-title">
            <Link to="/"><img src={logo} alt="logo" /></Link>
          </div>
          <nav>
            <ul>
              <li>
                <Link to="/sale-records">Records</Link>
              </li>
              <li>
                <Link to="/create-product">Create Product</Link>
              </li>
              <li>
                <Link to="/available-products.html">Available Products</Link>
              </li>
            </ul>
          </nav>
        </div>
      </header>
      <section id="product-area">
        <div className="container">
          <h1 id="product-header">Products</h1>
          <div className="products">
            {products === undefined ? <Spinner />
              : products.value.map(item => (
                <div className="product-item" key={item.id}>
                  <img src={item.product_image} className="product-image" />
                  <p className="product-name">{item.name}</p>
                  <p className="product-price">
#
                    {item.price}
                  </p>
                  <p>
                    <button className="editBtn"><Link to={`edit-product?id=${item.id}`}>Edit</Link></button>
                    {' '}
                    <button className="deleteBtn" type="submit" onClick={() => this.handleDelete(item.id)}>Delete</button>
                  </p>
                </div>
              ))}
          </div>
        </div>
      </section>
      <Footer />
    </div>
  );
}
}

const mapStateToProps = state => ({
  product: state.product,
  error: state.error,
});
export default connect(mapStateToProps, { getProducts, deleteProduct })(Admin);
