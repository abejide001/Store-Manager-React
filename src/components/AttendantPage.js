import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getProducts } from '../actions/productActions';
import Footer from './layouts/Footer';
import logo from '../assets/images/logo.png';
import '../assets/css/AttendantPage.css';
import '../assets/css/style.css';
import Spinner from '../common/Spinner';

export class AttendantPage extends Component {
  state = {
    search: '',
  };

  componentDidMount() {
    this.props.getProducts();
  }

  handleChange = (e) => {
    this.setState({ search: e.target.value });
  };

  render() {
    const { products } = this.props.product.products;
    const { search } = this.state;
    const foundProducts = products
      && products.value.filter(
        product => product.name.toLowerCase().includes(search.toLowerCase()),
      );

    return (
      <div>
        <header>
          <div className="container">
            <div id="header-title">
              <Link to="/">
                <img src={logo} alt="logo" />
              </Link>
            </div>
            <nav>
              <ul>
                <li>
                  <Link to="./available-products">Available Products</Link>
                </li>
                <li>
                  <Link to="./individual-attendant-sale-record">Records</Link>
                </li>
                <li>
                  <input
                    type="search"
                    placeholder="Search Product"
                    value={search}
                    name="search"
                    onChange={this.handleChange}
                  />
                </li>
              </ul>
            </nav>
          </div>
        </header>
        <section id="product-area">
          <div className="container">
            <h2>Products</h2>
            <div className="products">
              {foundProducts === undefined ? (
                <Spinner />
              ) : (
                foundProducts.map(item => (
                  <div className="product-item" key={item.id}>
                    <img src={item.product_image} className="product-image" />
                    <p className="product-name">{item.name}</p>
                    <p className="product-price">
#
                      {(item.price).toLocaleString()}
                                        </p>
                    <p>
                      <button className="product-button" type="button">
                        ADD TO CART
                      </button>
                    </p>
                  </div>
                ))
              )}
            </div>
          </div>
        </section>
        <section className="container cart-section">
          <div className="container">
            <h1 id="cart-title">CART</h1>
            <div className="cart-row">
              <span className="cart-item cart-header cart-column">ITEM</span>
              <span className="cart-price cart-header cart-column">PRICE</span>
              <span className="cart-quantity cart-header cart-column">
                QUANTITY
              </span>
            </div>
            <div className="cart-items" />
            <div className="cart-total">
              <strong className="cart-total-title">Total</strong>
              <span className="cart-total-price">#0</span>
            </div>
            <button className="purchase-button" type="button">
              PURCHASE
            </button>
          </div>
        </section>
        <Footer />
      </div>
    );
  }
}
const mapStateToProps = state => ({
  product: state.product,
});
export default connect(
  mapStateToProps,
  { getProducts },
)(AttendantPage);
