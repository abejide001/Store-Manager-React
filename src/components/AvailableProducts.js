/* eslint-disable jsx-a11y/alt-text */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getProducts } from '../actions/productActions';
import logo from '../assets/images/logo.png';
import '../assets/css/AvailableProducts.css';
import '../assets/css/style.css';
import Footer from './layouts/Footer';
import Spinner from '../common/Spinner';

export class Admin extends Component {
  componentWillMount() {
    this.props.getProducts();
  }

  render() {
    const { products } = this.props.product.products;
    return (
    <div>
      <header>
        <div className="container">
          <div id="header-title">
            <Link to="/"><img src={logo} alt="logo" /></Link>
          </div>
          <nav>
            <ul>
              <li>
                <Link to="/sale-records">Records</Link>
              </li>
              <li>
                <Link to="/add-product">Create Product</Link>
              </li>
              <li>
                <Link to="/available-products.html">Available Products</Link>
              </li>
            </ul>
          </nav>
        </div>
      </header>
      <section id="product-area">
        <div className="container">
          <h1 id="product-header">Available Products</h1>
          <div className="products">
            {products === undefined ? <Spinner />
              : products.value.map(item => (
                <div className="product-item" key={item.id}>
                  <img src={item.product_image} className="product-image" />
                  <p className="product-name">{item.name}</p>
                  <p className="product-price">
#
                    {item.price}
                  </p>
                  <p>
                    {' '}
                    <button className="modalBtn" type="submit">Details</button>
                  </p>
                </div>
              ))}
          </div>
        </div>
      </section>
      <Footer />
    </div>
  );
}
}

const mapStateToProps = state => ({
  product: state.product,
  error: state.error,
});
export default connect(mapStateToProps, { getProducts })(Admin);
