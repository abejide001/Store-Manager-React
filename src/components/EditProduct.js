/* eslint-disable react/forbid-prop-types */
/* eslint-disable no-undef */
/* eslint-disable camelcase */
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Footer from './layouts/Footer';
import { editProduct } from '../actions/productActions';
import logo from '../assets/images/logo.png';
import '../assets/css/EditProduct.css';

export class EditProduct extends Component {
    state = {
      name: '',
      price: '',
      quantity_in_inventory: '',
      product_image: '',
    }

    handleSubmit = (event) => {
      event.preventDefault();
      const regex = /\d+/g;
      const id = this.props.location.search.match(regex);
      const { name, price, quantity_in_inventory, product_image } = this.state;
      const productData = {
        name,
        price,
        quantity_in_inventory,
        product_image,
      };
      this.props.editProduct(id, productData);
    }

    handleChange = (e) => {
      this.setState({ [e.currentTarget.name]: e.currentTarget.value });
    };

    render() {
      const {
        name, price, quantity_in_inventory, product_image,
      } = this.state;
      return (
        <div>
          <div className="edit-body">
            <div className="edit-showcase">
              <header>
                <div className="container">
                  <div id="header-title">
                    <Link to="/">
                      <img src={logo} alt="logo" />
                    </Link>
                  </div>
                  <nav>
                    <ul>
                      <li>
                        <Link to="/admin">Admin</Link>
                      </li>
                    </ul>
                  </nav>
                </div>
              </header>
              <section id="edit">
                <form className="submission-form" onSubmit={this.handleSubmit}>
                  <h1>EDIT PRODUCT</h1>
                  <input type="text" className="input1" name="name" placeholder="Product Name" value={name} onChange={this.handleChange} required />
                  <input type="number" className="input2" name="price" placeholder="Price" value={price} onChange={this.handleChange} required />
                  <input type="number" className="input3" name="quantity_in_inventory" placeholder="quanity in inventory" value={quantity_in_inventory} onChange={this.handleChange} required />
                  <input type="url" name="product_image" placeholder="Image URL" value={product_image} onChange={this.handleChange} required />
                  <input type="submit" value="EDIT" id="sendBtn" />
                </form>
              </section>
              <Footer />
            </div>
          </div>
        </div>
      );
    }
}
EditProduct.propTypes = {
  location: PropTypes.object.isRequired,
  editProduct: PropTypes.func.isRequired,
};
const mapStateToProps = state => ({
  product: state.product,
});
export default connect(mapStateToProps, { editProduct })(EditProduct);
