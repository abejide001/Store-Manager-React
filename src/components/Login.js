/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/no-unused-prop-types */
/* eslint-disable import/no-webpack-loader-syntax */
/* eslint-disable react/forbid-prop-types */
/* eslint-disable comma-dangle */
/* eslint-disable react/no-access-state-in-setstate */
import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
// eslint-disable-next-line import/no-unresolved
import '!style-loader!css-loader!react-toastify/dist/ReactToastify.css';
import '../assets/css/Login.css';
import '../assets/css/style.css';
import logo from '../assets/images/logo.png';
import Footer from './layouts/Footer';
import { loginUser } from '../actions/authActions';

export class Login extends Component {
  state = {
    data: {
      email: '',
      password: '',
    },
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const user = {
      email: this.state.data.email,
      password: this.state.data.password,
    };
    this.props.loginUser(user, this.props.history);
  }

handleChange = (e) => {
  const data = { ...this.state.data };
  data[e.currentTarget.name] = e.currentTarget.value;
  this.setState({ data });
};

render() {
  const { email, password } = this.state.data;
  return (
    <div>
      <div className="login-body">
        <div className="login-showcase">
          <header>
            <div className="container">
              <div id="header-title">
                <Link to="/">
                  <img src={logo} alt="logo" />
                </Link>
              </div>
            </div>
          </header>
          <section id="login-area">
            <div className="container">
              <form onSubmit={this.handleSubmit}>
                <div className="login-box">
                  <h1>Login</h1>
                  <div className="login-textbox">
                    <i className="fas fa-envelope" />
                    <input type="email" placeholder="Email" name="email" onChange={this.handleChange} value={email} required />
                  </div>
                  <div className="login-textbox">
                    <i className="fas fa-key" aria-hidden="true" />
                    <input type="password" placeholder="Password" name="password" onChange={this.handleChange} value={password} required />
                  </div>
                  <button type="submit" className="btn">{this.props.auth.isLogging ? 'LOGGING IN...' : 'LOGIN'}</button>
                </div>
              </form>
            </div>
          </section>
        </div>
        <Footer />
      </div>
    </div>
  );
}
}
Login.propTypes = {
  loginUser: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  error: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired
};
const mapStateToProps = state => ({
  auth: state.auth,
  error: state.error,
});
export default connect(mapStateToProps, { loginUser })(withRouter(Login));
