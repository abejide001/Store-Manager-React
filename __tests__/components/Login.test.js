/* eslint-disable no-undef */
import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import { Login } from '../../src/components/Login';
import Footer from '../../src/components/layouts/Footer';

describe('<Login />', () => {
  describe('render()', () => {
    const props = {
      loginUser: jest.fn(),
      auth: jest.mock(),
      error: jest.mock(),
      history: { push: jest.fn() },
    };
    it('shouldn render the component', () => {
      const wrapper = shallow(<Login {...props} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should contain a Footer component ', () => {
      const wrapper = mount(<MemoryRouter><Login {...props} /></MemoryRouter>);
      expect(wrapper.find(Footer).length).toEqual(1);
    });
    it('has a text area user can type in ', () => {
      const wrapper = mount(<MemoryRouter><Login {...props} /></MemoryRouter>);
      wrapper.find('input').at(0).simulate('change', {
        target: { value: 'a' },
      });
      wrapper.update();
      expect(wrapper.find('input').at(0).prop('value')).toEqual('');
    });
    it('calls handleSubmit()', () => {
      const wrapper = shallow(<Login {...props} />);
      const event = Object.assign(jest.fn(), { preventDefault: () => {} });
      sinon.spy(wrapper.instance(), 'handleSubmit');
      wrapper.instance().handleSubmit(event);
      expect(wrapper.instance().handleSubmit.calledOnce)
        .toEqual(true);
      expect(wrapper.instance().handleSubmit.calledWith(event));
    });
  });
});
