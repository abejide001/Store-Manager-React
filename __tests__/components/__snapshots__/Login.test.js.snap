// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> render() shouldn render the component 1`] = `
<div>
  <div
    className="login-body"
  >
    <div
      className="login-showcase"
    >
      <header>
        <div
          className="container"
        >
          <div
            id="header-title"
          >
            <Link
              replace={false}
              to="/"
            >
              <img
                alt="logo"
                src="test-file-stub"
              />
            </Link>
          </div>
        </div>
      </header>
      <section
        id="login-area"
      >
        <div
          className="container"
        >
          <form
            className="login-form"
            onSubmit={[Function]}
          >
            <div
              className="login-box"
            >
              <h1>
                Login
              </h1>
              <div
                className="login-textbox"
              >
                <i
                  className="fas fa-envelope"
                />
                <input
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  type="email"
                  value=""
                />
              </div>
              <div
                className="login-textbox"
              >
                <i
                  aria-hidden="true"
                  className="fas fa-key"
                />
                <input
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                  value=""
                />
              </div>
              <input
                className="btn"
                type="submit"
                value="LOGIN"
              />
            </div>
          </form>
        </div>
      </section>
    </div>
    <Footer />
  </div>
</div>
`;
